features/bevragen/persoon/adressering/adres-regels/onbekendwaardes.feature

```diff
@@ -0,0 +1,53 @@
+
+  Abstract Scenario: persoon heeft een adres met <omschrijving>
+    Gegeven de persoon met burgerservicenummer '000000188' heeft de volgende 'verblijfplaats' gegevens
+    | functie adres (10.10) |
+    | W                     |
+    En de 'verblijfplaats' heeft de volgende 'adres' gegevens
+    | gemeentecode (92.10) | straatnaam (11.10) | huisnummer (11.20) |
+    | 0518                 | <straatnaam>       | <huisnummer>       |
+    Als personen wordt gezocht met de volgende parameters
+    | naam                | waarde                                                                                   |
+    | type                | RaadpleegMetBurgerservicenummer                                                          |
+    | burgerservicenummer | 000000188                                                                                |
+    | fields              | adressering.adresregel1,adressering.adresregel2,adressering.adresregel3,adressering.land |
+    Dan heeft de response een persoon zonder 'adressering' gegevens
+
+    Voorbeelden:
+    | straatnaam               | huisnummer | omschrijving                       |
+    |                          | 88         | een lege straatnaam                |
+    | .                        | 88         | een straatnaam met standaardwaarde |
+    | Jonkheer van Riemsdijkln | 0          | een huisnummer met standaardwaarde |
+
+Rule: adresregels voor een verblijfplaats buitenland worden niet geleverd als 'land (13.10)' de standwaarde heeft of 'regel 1 adres buitenland (13.30)', 'regel 2 adres buitenland (13.40)' en 'regel 3 adres buitenland (13.50)' hebben de standaardwaarde
+
+  Scenario: persoon heeft een verblijfplaats buitenland en 'land (13.10)' heeft de standaardwaarde
+    Gegeven de persoon met burgerservicenummer '000000346' heeft de volgende 'verblijfplaats' gegevens
+    | land (13.10) | regel 1 adres buitenland (13.30) | regel 2 adres buitenland (13.40) | regel 3 adres buitenland (13.50) |
+    | 0000         | Rue du pomme 26                  | Bruxelles                        | postcode 1000                    |
+    Als personen wordt gezocht met de volgende parameters
+    | naam                | waarde                                                                                   |
+    | type                | RaadpleegMetBurgerservicenummer                                                          |
+    | burgerservicenummer | 000000346                                                                                |
+    | fields              | adressering.adresregel1,adressering.adresregel2,adressering.adresregel3,adressering.land |
+    Dan heeft de response een persoon zonder 'adressering' gegevens
+
+  Scenario: persoon heeft een verblijfplaats buitenland en 'regel 1 adres buitenland (13.30)', 'regel 2 adres buitenland (13.40)' en 'regel 3 adres buitenland (13.50)' hebben de standaardwaarde
```

Comment:
ik denk dat er vanuit de logica hier 2 situaties zijn die moeten worden afgevangen:
1. het land is onbekend: eventuele waarden voor regel 1, 2 en 3 is dan niet meer relevant(voor onderscheid met situatie 2 geef je die wel een waarde)
2. alleen land is onbekend: land heeft een waarde≠standaardwaarde en regel 1, regel 2 én regel 3 hebben geen waarde (null of lege string)

De eerste situatie wordt getoond met Scenario: persoon heeft een verblijfplaats buitenland en 'land (13.10)' heeft de standaardwaarde

Dit tweede scenario test ook de eerste situatie, want land heeft de standaardwaarde.

Ik zou hier dus liever een scenario zien die toont dat land een waarde≠0000 heeft

Reply:
aangepast